### PR TITLE
feat:  integrate clip and model evaluation into cross modal search

### PR DESCRIPTION
**Pull request by bwanglzu**
Please checkout the description in issue

### CODE REVIEW

File: cross-modal-search/evaluate.py [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+        os.environ['JINA_IMAGE_ENCODER'] = 'docker://jinahub/pod.encoder.vseimageencoder:0.0.5-1.0.7'
+        os.environ['JINA_TEXT_ENCODER'] = 'docker://jinahub/pod.encoder.vsetextencoder:0.0.6-1.0.7'
+        os.environ['JINA_TEXT_ENCODER_INTERNAL'] = 'yaml/vse/text-encoder.yml'
+    else:
+        msg = f'Unsupported model {model_name}.'
+        msg += 'Expected `clip` or `vse`.'
+        raise ValueError(msg)
+
+
+def input_index_data(num_docs=None, batch_size=8, dataset_type='f30k'):
```
Comment (JoanFM) [inline cross-modal-search/evaluate.py:32]: we could maybe move all these generators into `dataset.py`?

File: cross-modal-search/flow-index.yml [hunk 2]
```diff
@@ -26,8 +26,8 @@ pods:
     shards: $JINA_SHARDS
     needs: [gateway]
   - name: text_encoder
-    uses: $JINA_USES_VSE_TEXT_ENCODER
-    uses_internal: yaml/text-encoder.yml
+    uses: $JINA_TEXT_ENCODER
+    uses_internal: $JINA_TEXT_ENCODER_INTERNAL
```
Comment (JoanFM) [inline cross-modal-search/flow-index.yml:30]: I comment it here, but is not here, change the `_merge` in line 40 for `_pass`

File: cross-modal-search/evaluate.py [hunk 3]
```diff
@@ -88,11 +91,14 @@ def evaluation_generator(num_docs=None, batch_size=8, dataset_type='f8k'):
...
     for doc in resp.search.docs:
-        sum_of_score += doc.evaluations[0].value
-        print(f'Evaluation {doc.evaluations[0].op_name}: {doc.evaluations[0].value}')
-    print(f'MeanReciprocalRank is: {sum_of_score/len(resp.search.docs)}')
+        batch_of_score += doc.evaluations[0].value
+    print(f'MeanReciprocalRank of the batch is: {batch_of_score/len(resp.search.docs)}')
+    global sum_of_score
+    global num_of_matches
+    sum_of_score += batch_of_score
+    num_of_matches += len(resp.search.docs)
```
Comment (JoanFM) [inline cross-modal-search/evaluate.py:101]: as a sanity check, can u sum the num of matches naively from the number of Evaluations u compute and see if it matches

File: cross-modal-search/flow-index.yml (hunk 2 above)
Comment (JoanFM) [reply cross-modal-search/flow-index.yml:30]: my bad, better to keep it as `_merge_root`

File: cross-modal-search/README.md [hunk 4]
```diff
@@ -38,49 +38,40 @@ In this example, `jina` is used to implement a Cross-modal search system. This e
...
 
-The models used for the example are cited from the paper _Improving Visual-Semantic Embeddings with Hard Negatives_ (https://arxiv.org/pdf/1707.05612.pdf).
+The models used for the example are cited from the paper, you can try our example with one of them:
 
-To make this search system retrieve good results, we have used the models trained in [https://github.com/fartashf/vsepp](https://github.com/fartashf/vsepp) . A model has been trained that encodes `text` and `images` in a common embedding space trying to put together the embedding of images and its corresponding captions.
+1. [CLIP: Contrastive Language-Image Pre-Training](https://arxiv.org/abs/2007.13135) (recommend)
+2. [VSE++: Improving Visual-Semantic Embeddings with Hard Negatives](https://arxiv.org/pdf/1707.05612.pdf).
 
-We use one network per modality:
+Both of the models has been trained that encodes pair of `text` and `images` into a common embedding space.
```
Comment (JoanFM) [inline cross-modal-search/README.md:64]: both of the models have been trained to encode

**Review by JoanFM — CHANGES_REQUESTED** [review]

File: cross-modal-search/README.md [hunk 5]
```diff
@@ -61,7 +61,7 @@ The models used for the example are cited from the paper, you can try our exampl
 1. [CLIP: Contrastive Language-Image Pre-Training](https://arxiv.org/abs/2007.13135) (recommend)
 2. [VSE++: Improving Visual-Semantic Embeddings with Hard Negatives](https://arxiv.org/pdf/1707.05612.pdf).
 
-Both of the models has been trained that encodes pair of `text` and `images` into a common embedding space.
+Both of the models have been trained that encodes pair of `text` and `images` into a common embedding space.
```
Comment (JoanFM) [inline cross-modal-search/README.md:64]: have been trained 'to' encode

**Comment by bwanglzu** [comment]
waiting for `1.0.8` release

**Comment by JoanFM** [comment]
> waiting for `1.0.8` release

we need to find a deterministic hashing where we can provide the  seed or something

File: cross-modal-search/dataset.py [hunk 6]
```diff
@@ -106,12 +108,12 @@ def input_index_data(num_docs=None, batch_size=8, dataset_type='f30k'):
 
     for i, (images, captions) in enumerate(data_loader):
         for image, caption in zip(images, captions):
-            current_hash = hash(image)
+            encoded = base64.b64encode(image)
```
Comment (JoanFM) [inline cross-modal-search/dataset.py:111]: does this depend on thr PYTHONSEED as well?

**Comment by bwanglzu** [comment]
@JoanFM no, i forgot to remove this

**Review by JoanFM — APPROVED** [review]

**Pull request merged**